block/src/qcow/mod.rs

```diff
@@ -30,105 +30,92 @@ use vmm_sys_util::{
 pub use crate::qcow::raw_file::RawFile;
 
 #[sorted]
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
+    #[error("backing file io error: {0}")]
     BackingFileIo(io::Error),
+    #[error("backing file open error: {0}")]
     BackingFileOpen(Box<crate::Error>),
+    #[error("backing file name is too long: {0} bytes over")]
     BackingFileTooLong(usize),
+    #[error("compressed blocks not supported")]
     CompressedBlocksNotSupported,
+    #[error("failed to evict cache: {0}")]
     EvictingCache(io::Error),
+    #[error("file larger than max of {}: {0}", MAX_QCOW_FILE_SIZE)]
     FileTooBig(u64),
+    #[error("failed to get file size: {0}")]
     GettingFileSize(io::Error),
+    #[error("failed to get refcount: {0}")]
     GettingRefcount(refcount::Error),
+    #[error("failed to parse filename: {0}")]
     InvalidBackingFileName(str::Utf8Error),
+    #[error("invalid cluster index")]
     InvalidClusterIndex,
+    #[error("invalid cluster size")]
     InvalidClusterSize,
+    #[error("invalid index")]
     InvalidIndex,
+    #[error("invalid L1 table offset")]
     InvalidL1TableOffset,
+    #[error("invalid L1 table size: {0}")]
     InvalidL1TableSize(u32),
+    #[error("invalid magic")]
     InvalidMagic,
+    #[error("invalid offset")]
     InvalidOffset(u64),
+    #[error("invalid refcount table offset")]
     InvalidRefcountTableOffset,
+    #[error("invalid refcount table size: {0}")]
     InvalidRefcountTableSize(u64),
+    #[error("no free clusters")]
     NoFreeClusters,
+    #[error("no refcount clusters")]
     NoRefcountClusters,
+    #[error("not enough space for refcounts")]
     NotEnoughSpaceForRefcounts,
+    #[error("failed to open file {0}")]
     OpeningFile(io::Error),
+    #[error("failed to read data: {0}")]
     ReadingData(io::Error),
+    #[error("failed to read header: {0}")]
     ReadingHeader(io::Error),
+    #[error("failed to read pointers: {0}")]
```

Comment:
Most of the messages start with a capital letter - i.e. sentence case - thee should be the same.